### PR TITLE
scheduled job to run at specific times

### PR DESCRIPTION
**Pull request by AnointingMax**

### CODE REVIEW

**Review by LeanKhan — COMMENTED** [review]
Looks good to me sha, but I don't know how I can test it without changing the interval time. Unless I just do that sha

File: app/routes.py [hunk 1]
```diff
@@ -377,31 +378,46 @@ def remove_candidate():
 def missing_route():
     return render_template("404.html")
 
-
-# the update status function is the function to be ran repaeatedly hence the id
-# the function will be ran in 30 seconds intervals
-@scheduler.task('interval', id='do_update_status', seconds=10, misfire_grace_time=900)
-def update_status():
-    now = datetime.now()
+def schedule_job(id):
```
Comment (danieliheonu) [inline app/routes.py:394]: Could you add this method to tools.py?  Let's keep routes.py strictly for routes

File: app/routes.py [hunk 2]
```diff
@@ -377,31 +378,46 @@ def remove_candidate():
...
-                    #check if the current time is greater than the starting time of the election if yes set the status of the election to started
-                    if election.starting_at < now:
-                        election.status_id = 2
-                # check if the status of the election is started
-                elif election.status == statuses[1]:
-                    #check if the current time is greater than the ending time of the election if yes set the status of the election to ended
-                    if election.ending_at < now:
-                        election.status_id = 3
+                # schedule them to run the schuled_job method at their starting datetime
+                scheduler.add_job(id="start"+election.name, func=schedule_job, trigger='date', run_date=election.starting_at, args=[election.id])
```
Comment (LeanKhan) [inline app/routes.py:419]: Nice work, I'm just wondering why we can't add these jobs as soon as we create the Election?

File: app/routes.py [hunk 3]
```diff
@@ -377,31 +378,46 @@ def remove_candidate():
...
-                    if election.ending_at < now:
-                        election.status_id = 3
+                # schedule them to run the schuled_job method at their starting datetime
+                scheduler.add_job(id="start"+election.name, func=schedule_job, trigger='date', run_date=election.starting_at, args=[election.id])    
             except Exception as e:
                 print(str(e))
+
+
+# task is to run at the beginning of everyday
+@scheduler.task('date', id='update_status_for_started', run_date=datetime.now())
```
Comment (LeanKhan) [inline app/routes.py:425]: Why is this one `run_date=datetime.now()` and not `days="*"` like the first one for pending?

File: app/routes.py (hunk 1 above)
Comment (LeanKhan) [reply app/routes.py:394]: Or better still, a different file for our jobs. maybe `jobs.py` or something

**Review by LeanKhan — APPROVED** [review]
I tested it by changing the times and everything works. I just wondered if we can create the jobs as soon as the Election is being created, so we don't have to run that `cron` or `date` jobs.

File: app/routes.py (hunk 2 above)
Comment (AnointingMax) [reply app/routes.py:419]: I thought of doing that but nothing came to mind on how to do it

File: app/routes.py (hunk 3 above)
Comment (AnointingMax) [reply app/routes.py:425]: Sorry that's a mistake its meant to be days='*'

**Pull request merged**